### ISSUE TITLE
Input value cannot be rendered

### ISSUE THREAD

**Issue by liming** [WebReflection/hyperHTML#213]
Have a look at this simple example:
https://codepen.io/hellosleep/pen/jzVvjd

```js
const defaultValue = 10;

hyperHTML.bind(document.querySelector('#app'))`
  <input type="number" value=${defaultValue}/>
  <input type="number" value="10"/>
`;
```

Here I want to output the input value to display an input with a value, which does not display at all in the HTML. The output of the code are:

```html
<input type="number">
<input type="number" value="10">
```

The value is not rendered so it cannot be printed out.

**Comment by WebReflection** [comment]
> The value is not rendered so it cannot be printed out.

what do you mean? I can both read the value and if I print the page I can see the number too.

Please note, if you focus on the produced _hyperHTML_ output you'll find a lot of things you should never care at all and none of this will ever be considered bugs, these are how the engine works, it abstracts away thing for you.

TL;DR it's pointless to inspect the produced HTML, there's nothing interesting there 😉

**Comment by liming** [comment]
@WebReflection 

There is a problem. I made another sample. You can see the options value has been output as html, but input's value is not. 

Think about a use case that we need output a element's innerHTML. The HTML does not contain value="10", so it cannot be displayed. 

https://codepen.io/hellosleep/pen/jzVvjd?editors=0010

```js
const defaultValue = 10;

hyperHTML.bind(document.querySelector('#app'))`
  <input type="number" value=${defaultValue}/>
  <input type="number" value="10"/>
  <select>
    <option value=${defaultValue}/>
    <option value="20"/>
  </select>
`;
```

**Comment by liming** [comment]
@WebReflection 

Here is the copy outerHTML. You can see the problem clearly this way.

```html
<div id="app">
  <input type="number" max="100">
  <input type="number" max="100" value="10">
  <select>
    <option value="10"></option>
    <option value="20"></option>
  </select>
</div>
```

**Comment by WebReflection** [comment]
> Think about a use case that we need output a element's innerHTML

The element innerHTML is the template you pass, not the output it has.

This cannot possibly work for various reasons:

  * you can pass callbacks or objects as listeners, it's pointless to have these in the layout
  * you can pass `checked=${false}` which obviously won't be in the layout
  * you can pass `disabled=${true}` which is handled internally

Direct attributes also for compatibility, reliability, and performance reasons, are removed and never set again in the node because if you have an input element and later on you do `input.setAttribute("value", "123")` you won't see that reflected in the view (with or without hyperHTML).

The layout hyperHTML produces is no 3rd parts business. hyperHTML own its layout and does things to make its mechanism work. If you need to represent a clean HTML from a generic page/element you need a proper HTML sanitizer/parser that understands elements and cares about special accessors such `value` in `input` or `src` in `images` etc.

This is what _printing page_ does, as example, and indeed you see all the values you need.

TL;DR due DOM constrains with builtins accessors, there is no way this would be changed in here.

**Comment by liming** [comment]
Hi @WebReflection , fair enough. I have found the way around to manually set the attributes.
